allennlp/nn/beam_search.py

```diff
@@ -381,3 +409,178 @@ def _search(
         all_predictions = torch.cat(list(reversed(reconstructed_predictions)), 2)
 
         return all_predictions, last_log_probabilities
+
+    @classmethod
+    def without_sampling(
+        cls,
+        end_index: int,
+        max_steps: int = 50,
+        beam_size: int = 10,
+        per_node_beam_size: int = None,
+        sampler: Sampler = None,
```

Comment:
If we're calling this method `without_sampling` then we should probably not allow a `sampler`.
```suggestion
```